tests/unit/test_disable_compilation.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+from spyre_util import write_sample_model_config
+from vllm.config import (ModelConfig, ParallelConfig, SchedulerConfig,
+                         VllmConfig)
+
+from vllm_spyre.platform import (PRE_COMPILE_MODEL_CATALOG_FILENAME,
+                                 SpyrePlatform)
+
+
+@pytest.mark.parametrize("batch_type", ["sb", "cb"])
+def test_handle_disable_compilation(monkeypatch, tmp_path, batch_type):
+    """
+    Test _handle_disable_compilation for static and continuous batching.
+    Note: since the validation here is only giving warning in case of mismatch,
+    these tests will only fail if there is a bug in the logic
+    """
+
+    # Patch version to avoid test failures around version mismatch
+    monkeypatch.setattr("vllm_spyre._version.version", "0.8.0")
+
+    if batch_type == "sb":
+        monkeypatch.setenv("VLLM_SPYRE_WARMUP_PROMPT_LENS", "128")
+        monkeypatch.setenv("VLLM_SPYRE_WARMUP_NEW_TOKENS", "128")
+        monkeypatch.setenv("VLLM_SPYRE_WARMUP_BATCH_SIZES", "1")
+        monkeypatch.setenv("VLLM_SPYRE_USE_CB", "0")
+        sample_model_config = {
+            "vllm_spyre_version": "0.8.0",
+            "data": {
+                "MODEL_NAME": "/models/granite-3.3-8b-instruct-FP8",
+                "NUM_AIUS": 2,
+                "VLLM_SPYRE_WARMUP_PROMPT_LENS": "128",
+                "VLLM_SPYRE_WARMUP_NEW_TOKENS": "128",
+                "VLLM_SPYRE_WARMUP_BATCH_SIZES": "1",
+            },
+        }
+
+    else:
+        monkeypatch.setenv("VLLM_SPYRE_USE_CB", "1")
+        sample_model_config = {
+            "vllm_spyre_version": "0.8.0",
+            "data": {
+                "MODEL_NAME": "/models/granite-3.3-8b-instruct-FP8",
+                "NUM_AIUS": 2,
+                "VLLM_DT_MAX_CONTEXT_LEN": 256,
+                "VLLM_DT_MAX_BATCH_SIZE": 2
+            },
+        }
+
+    write_sample_model_config(tmp_path, sample_model_config)
+
+    monkeypatch.setenv("DISABLE_COMPILATION", "true")
+    monkeypatch.setenv("TORCH_SENDNN_CACHE_DIR", str(tmp_path))
+    monkeypatch.setenv("TORCH_SENDNN_CACHE_ENABLE", "1")
+
+    vllm_config = VllmConfig(
+        model_config=ModelConfig(
+            model="ibm-ai-platform/micro-g3.3-8b-instruct-1b",
+            max_model_len=256),
+        parallel_config=ParallelConfig(tensor_parallel_size=2),
+        scheduler_config=SchedulerConfig(max_num_seqs=2))
+
+    SpyrePlatform._handle_disable_compilation(vllm_config, is_decoder=True)
```

Comment:
Going off of Travis' finding that we call `check_and_update_config` anyway in the post init for `VllmConfig`, can we not omit this explicit call?

Reply:
I think we could, but then we should add a way to assert that the `_handle_disable_compilation` function is actually called in each test where we expect it to be.

I also looked in to creating an actual dummy VllmConfig that didn't do all of the `__post_init__` that we could pass to `_handle_disable_compilation`, but hit a Pydantic error if the ModelConfig wasn't post-init'd...

Reply:
Also I was able to get warnings asserted using custom caplog! That way we can test the actual warning message and omit the duplicate call to disable compilation

Reply:
Also in my testing I had to monkeypatch `REQUIRE_PRECOMPILED_DECODERS` to True otherwise the code returns immediately after this line without actually running through the entire code:
```
if not envs_spyre.VLLM_SPYRE_REQUIRE_PRECOMPILED_DECODERS:
        return
```